vim/core/oni-core-statusbar/index.js

```diff
@@ -4,10 +4,10 @@ const rgb = (r, g, b) => `rgb(${r}, ${g}, ${b})`
 
 const activate = (Oni) => {
     const React = Oni.dependencies.React
-
-    const workingDirectoryItem = Oni.statusBar.createItem(0, -1, "oni.status.workingDirectory")
-    const lineNumberItem = Oni.statusBar.createItem(1, -1, "oni.status.lineNumber")
-    const modeItem = Oni.statusBar.createItem(1, -2, "oni.status.mode")
+    const { workingDirectory = 0, linenumber = 0, mode = 0 } = Oni.configuration.getValue("statusbar.priority")
+    const workingDirectoryItem = Oni.statusBar.createItem(0, workingDirectory, "oni.status.workingDirectory")
```

Comment:
If we're making the priority configurable - maybe we don't even need to pass the priority in as part of the API now? If the configuration is the source of truth, maybe we could just use that. In addition, we could leverage the 'id' that we pass in here - like `statusbar.priority': { "oni.status.workingDirectory": 1}`. The nice thing about that is it makes it easy for plugins to then also be prioritized in the same system.

Reply:
@bryphe do you mean deleting it from the createItem method, and the api package and then adding the priority in the statusbar component somewhere like get config value etc.? (for clarification) ^^